Cart Flow/Custom Views/LocationSelectionViews/AisleSelectionView.swift

```diff
@@ -0,0 +1,173 @@
+    
+    let aislesLabel = CFSecondaryTitleLabel()
+    let aisleDescriptionLabel = CFSecondaryTitleLabel()
+    let plusButton = CFPlusButton()
+    let minusButton = CFMinusButton()
+    
+    let aislePicker = UIPickerView()
+    
+    var aisles: [Aisle] = []
+    var selectedStore: GroceryStore? = nil
+    var selectedAisle: Aisle? = nil
+    var rotationAngle: CGFloat!
+    
+    weak var delegate: AisleSelectionViewDelegate!
+    
+    
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        
+        configureAislePicker()
+        layoutUI()
+    }
+    
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    
+    private func configureAislePicker() {
+        self.aislePicker.delegate = self
+        self.aislePicker.dataSource = self
+    }
+    
+    
+    private func layoutUI() {
+        addSubview(aislesLabel)
+        addSubview(aisleDescriptionLabel)
+        addSubview(aislePicker)
+        
+        let padding: CGFloat = 15
+        rotationAngle = 90 * (.pi/180)
+        
+        
+        aislesLabel.text = "Aisle:"
+        aislesLabel.font = UIFont.boldSystemFont(ofSize: 18)
+        
+        aislePicker.setValue(UIColor.black, forKey: "textColor")
+        aislePicker.transform = CGAffineTransform(rotationAngle: rotationAngle)
+        aislePicker.translatesAutoresizingMaskIntoConstraints = false
+        
+        NSLayoutConstraint.activate([
+            aislesLabel.topAnchor.constraint(equalTo: self.topAnchor, constant: padding),
+            aislesLabel.leadingAnchor.constraint(equalTo: self.leadingAnchor, constant: padding),
+            aislesLabel.widthAnchor.constraint(equalToConstant: 55),
+            aislesLabel.heightAnchor.constraint(equalToConstant: 26),
+            
+            aisleDescriptionLabel.topAnchor.constraint(equalTo: self.topAnchor, constant: padding),
+            aisleDescriptionLabel.leadingAnchor.constraint(equalTo: aislesLabel.trailingAnchor, constant: padding),
+            aisleDescriptionLabel.trailingAnchor.constraint(equalTo: self.trailingAnchor, constant: -padding),
+            aisleDescriptionLabel.heightAnchor.constraint(equalToConstant: 26),
+            
+            aislePicker.widthAnchor.constraint(equalToConstant: 40),
+            aislePicker.heightAnchor.constraint(equalToConstant: 400),
+            aislePicker.centerYAnchor.constraint(equalTo: aislesLabel.bottomAnchor, constant: padding),
+            aislePicker.centerXAnchor.constraint(equalTo: self.centerXAnchor)
+        ])
+    }
+    
+    
+    func setLocationToEdit(aisle: Aisle) {
+        
+        if let aisleToEdit = aisles.firstIndex(of: aisle) {
+            print("aisle: \(aisleToEdit)")
+            aislePicker.selectRow((aisleToEdit), inComponent: 0, animated: false)
+        }
+    }
+    
+    
+    func loadAisles() {
```

Comment:
🚨Accessing your model in a view breaks the MVC contract.🚨

It looks like you went with the MVC architecture for this app (https://www.raywenderlich.com/1000705-model-view-controller-mvc-in-ios-a-modern-approach). In MVC, you should never access your model directly from your views. In this case, it looks like you file was originally a `UIViewController` that you converted to a `UIView`. I'm assuming you did this so you can embed this view as needed. I'd suggest subclassing `UIViewController` here instead of `UIView` and then embedding this view controller as a child of `LocationSelectionView` (which itself should also be a view controller). More on embedding child view controllers here: https://www.swiftbysundell.com/basics/child-view-controllers/.

Alternatively, you could keep these classes as `UIView`s, and have your `AddNewItemVC` class fetch the data from the model and pass it to the `LocationSelectionView`, which could pass them to the `AisleSelectionView`.

But either way, you don't want to access your model from your view code, controllers are there to form the relationship between the two.

Reply:
Was wondering about this (specifically loading the model from a view). Definitely tried doing it as a VC at first. I don't remember exactly why I bailed on it for UIView, but I've learned a good bit more since I first made that section so I'll go back and refactor accordingly!